sklearnex/spmd/basic_statistics/tests/test_incremental_basic_statistics_spmd.py

```diff
@@ -260,6 +271,7 @@ def test_incremental_basic_statistics_partial_fit_spmd_synthetic(
     split_data = np.array_split(data, num_blocks)
 
     if weighted:
+        # Create weights array containing the weight for each sample in the data
```

Comment:
Why is 0-7 chosen? Its clear you are adding a weight for each sample in the data, but why are those values chosen? @olegkkruglov

Reply:
do you have better option? it's just an example, why did this even catch your attention? does that mean that there are no more serious issues?

Reply:
If its this, then I think we may be misusing it: https://www.testenvironmentmanagement.com/what-are-test-data-gold-copies/

Reply:
for me gold data is just small dataset making the final answer and intermediate results relatively easy to calculate and verify separately and independently in case if something is going wrong

Reply:
To me it seems fine for this PR. To be more "gold" maybe it would make sense to use an array of 0s and 1s for simplicity but this should also be fine. A broader refactor of spmd algos can be done after the incrementals are merged where we centralize some dataset generation logic but as a standalone PR I have no objections.